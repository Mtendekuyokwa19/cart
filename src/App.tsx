import React from 'react'
import { RouteSet } from './route'
import "./index.css"
export default function App() {
  return (

  <RouteSet/>
)
}

