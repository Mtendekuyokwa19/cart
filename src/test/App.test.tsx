
// App.test.jsx
import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import App from "../App";
import userEvent from "@testing-library/user-event";
describe("App component", () => {
  it("renders correct heading", () => {
    const{container}=render(<App />);
    expect(container).toMatchSnapshot();
  }


  );
 it("clicke buttone",async()=>{

    const user=userEvent.setup()
    render(<App/>)
   await user.click(screen.getByTestId("button")) 
    let header=screen.getByRole("heading");
    expect(header.textContent).toMatch(/Radical Rhinos/i)

  }) 
});

