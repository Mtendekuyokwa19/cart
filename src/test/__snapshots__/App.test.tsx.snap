// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders correct heading 1`] = `
<div>
  <button
    data-testid="button"
    type="button"
  >
    Click Me
  </button>
  <h1>
    Magnificent Monkeys
  </h1>
</div>
`;
